### PR TITLE
e2e: Provide e2e config yaml to web3signer

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

The end to end testing uses custom config files for network testing. This PR introduces the custom e2e_config.yaml and enforces the hardcoded values match the yaml. This PR also provides the e2e_config.yaml to the web3signer.

**Which issues(s) does this PR fix?**

(Probably) Unblocks e2e issues in https://github.com/prysmaticlabs/prysm/pull/10056

**Other notes for review**

Blocked by https://github.com/ConsenSys/web3signer/issues/494

This can also be used to fix https://github.com/prysmaticlabs/prysm/pull/9974

### CODE REVIEW

**Comment by prestonvanloon** [comment]
This is still blocked until web3signer & teku issue another release. Unless we can find a way to build from a specific commit with web3signer and Teku. I've pinged the web3signer channel on discord to ask how/if we can do this.

**Comment by prestonvanloon** [comment]
This can be made ready for review after #10281 merges.

File: testing/endtoend/minimal_e2e_test.go
Comment (prestonvanloon) [inline testing/endtoend/minimal_e2e_test.go:30]:
```suggestion
```

**Comment by prestonvanloon** [comment]
This seems to pass consistently in CI.

File: testing/endtoend/deps.bzl
Comment (prestonvanloon) [inline testing/endtoend/deps.bzl:10]:
```suggestion
        # Built from commit 17d253b which has important unreleased changes.
```

File: testing/endtoend/minimal_e2e_test.go
Comment (prestonvanloon) [inline testing/endtoend/minimal_e2e_test.go:27]:
```suggestion
```

File: testing/endtoend/minimal_e2e_test.go
Comment (prestonvanloon) [inline testing/endtoend/minimal_e2e_test.go:38]:
```suggestion
// Run minimal e2e config with the current release validator against latest beacon node.
```

File: testing/endtoend/endtoend_test.go
Comment (prestonvanloon) [inline testing/endtoend/endtoend_test.go:93]:
```suggestion
```

**Review by rauljordan — APPROVED** [review]

**Review by james-prysm — COMMENTED** [review]
There were a few different things to fix with this PR when I got it, 
- There were a lot of **port in use** errors while running locally. **THESE ARE EXISTING ISSUES** and should be addressed separately, this threw off the investigation.
- the first was matching the versions of the byte to hex in the yml file vs the e2e configs. the e2e configs uses a byte view of the fork versions while the yaml uses a hex view. this was breaking the unit tests.
- web3signer's tagged version doesn't support the changes to allow to override the network configs correctly. **we are using the latest develop version in this PR and will need to keep in mind to update in the future!** we assumed maybe it was something broken with web3signer and so we tried to update this.
- Web3signer does not support bellatrix as of this PR and changes to the yaml config / a special condition check in the e2e minimal test were added to address this. the web3signer e2e will only support Altair for now. This also addressed some differences between the yaml file configurations with the configuration file from our e2e config.
- The main issue was a code issue found while investigating this error ( after we fixed the other config problems and so on)
`{"type":"object","properties":{"contribution_and_proof":{"$ref":"urn:vertxschemas:bdbd8f25-a36f-4a7b-8a24-69008737b2bd#"}},"required":["contribution_and_proof"]} and input {"type":"BLOCK_V2","fork_info":{"fork":{"previous_version":"0x000000fd","current_version":"0x010000fd","epoch":"1"},"genesis_validators_root":"0x5e09135edfe369e2677ea042da32079be461fe30eee9bac733652fe630caa4ff"},"signingRoot":"0x81eafbb2aef458907f23d00feddbb0fde965b011a8791480373edb50eb9a2ceb","beacon_block":{"version":"ALTAIR","beacon_block"`
It was strange at first that it was looking for contribution_and_proof in this object, but in reality the input had one off property so web3signer's code went through all the known schemas validations with contribution_and_proof being the last one and there by throwing this error. details on the fix found here #10490 
- there was a merge issue reverting some of the code needed for the JWT token, because this PR was so old it merged incorrectly overwriting the correct code, this was reverted to fix the following error 
`time="2022-04-06 20:29:15" level=error msg="could not read JWT secret file for authenticating execution API: open /root/.cache/bazel/_bazel_root/3c0796cd90bdf40c3e7dfd9539754070/sandbox/processwrapper-sandbox/1536/execroot/prysm/_tmp/bbead6fe9ddf0afc2b3e673ead7d16a4/shard-0/eth1data/2/geth/jwtsecret: no such file or directory" prefix=main`

**Pull request merged**
